services/debugger/uploader.go

```diff
@@ -210,3 +221,20 @@ func (uploader *uploaderImpl[E]) flushEvents(ctx context.Context) {
 		}
 	}
 }
+
+func getAuthUsername() (string, error) {
```

Comment:
Consider moving this auth logic to a different component. deploymentType logic should remain contained in a few components. Can we use Identify method from backend config ? https://github.com/rudderlabs/rudder-server/blob/master/backend-config/namespace_config.go#L228C27-L228C38

Reply:
Fixed